flaky `TestQuerier_ValidateQueryTimeRange_MaxQueryLength`

[CI log](https://github.com/grafana/mimir/actions/runs/12129293054/job/33817446378)

```
--- FAIL: TestQuerier_ValidateQueryTimeRange_MaxQueryLength (0.00s)
    --- FAIL: TestQuerier_ValidateQueryTimeRange_MaxQueryLength/should_forbid_query_on_large_time_range_over_the_limit_and_short_rate_time_window (0.00s)
        querier_test.go:[80](https://github.com/grafana/mimir/actions/runs/12129293054/job/33817446378#step:8:81)2: 
            	Error Trace:	/__w/mimir/mimir/pkg/querier/querier_test.go:802
            	Error:      	Not equal: 
            	            	expected: "expanding series: the query time range exceeds the limit (query length: 721h0m59.999s, limit: 720h0m0s) (err-mimir-max-query-length). To adjust the related per-tenant limit, configure -querier.max-partial-query-length, or contact your service administrator."
            	            	actual  : "expanding series: the query time range exceeds the limit (query length: 721h1m0s, limit: 720h0m0s) (err-mimir-max-query-length). To adjust the related per-tenant limit, configure -querier.max-partial-query-length, or contact your service administrator."
            	            	
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -1 +1 @@
            	            	-expanding series: the query time range exceeds the limit (query length: 721h0m59.999s, limit: 720h0m0s) (err-mimir-max-query-length). To adjust the related per-tenant limit, configure -querier.max-partial-query-length, or contact your service administrator.
            	            	+expanding series: the query time range exceeds the limit (query length: 721h1m0s, limit: 720h0m0s) (err-mimir-max-query-length). To adjust the related per-tenant limit, configure -querier.max-partial-query-length, or contact your service administrator.
            	Test:       	TestQuerier_ValidateQueryTimeRange_MaxQueryLength/should_forbid_query_on_large_time_range_over_the_limit_and_short_rate_time_window
FAIL
FAIL	github.com/grafana/mimir/pkg/querier	[82](https://github.com/grafana/mimir/actions/runs/12129293054/job/33817446378#step:8:83).780s
```

--- a/pkg/querier/querier_test.go
+++ b/pkg/querier/querier_test.go
@@ -735,6 +735,7 @@ func TestQuerier_ValidateQueryTimeRange(t *testing.T) {
 
 func TestQuerier_ValidateQueryTimeRange_MaxQueryLength(t *testing.T) {
 	const maxQueryLength = 30 * 24 * time.Hour
+	now := time.Now()
 
 	tests := map[string]struct {
 		query          string
@@ -744,26 +745,26 @@ func TestQuerier_ValidateQueryTimeRange_MaxQueryLength(t *testing.T) {
 	}{
 		"should allow query on short time range and rate time window close to the limit": {
 			query:          "rate(foo[29d])",
-			queryStartTime: time.Now().Add(-time.Hour),
-			queryEndTime:   time.Now(),
+			queryStartTime: now.Add(-time.Hour),
+			queryEndTime:   now,
 			expected:       nil,
 		},
 		"should allow query on large time range close to the limit and short rate time window": {
 			query:          "rate(foo[1m])",
-			queryStartTime: time.Now().Add(-maxQueryLength).Add(time.Hour),
-			queryEndTime:   time.Now(),
+			queryStartTime: now.Add(-maxQueryLength).Add(time.Hour),
+			queryEndTime:   now,
 			expected:       nil,
 		},
 		"should forbid query on short time range and rate time window over the limit": {
 			query:          "rate(foo[31d])",
-			queryStartTime: time.Now().Add(-time.Hour),
-			queryEndTime:   time.Now(),
+			queryStartTime: now.Add(-time.Hour),
+			queryEndTime:   now,
 			expected:       errors.Errorf("expanding series: %s", NewMaxQueryLengthError(745*time.Hour-time.Millisecond, 720*time.Hour)),
 		},
 		"should forbid query on large time range over the limit and short rate time window": {
 			query:          "rate(foo[1m])",
-			queryStartTime: time.Now().Add(-maxQueryLength).Add(-time.Hour),
-			queryEndTime:   time.Now(),
+			queryStartTime: now.Add(-maxQueryLength).Add(-time.Hour),
+			queryEndTime:   now,
 			expected:       errors.Errorf("expanding series: %s", NewMaxQueryLengthError((721*time.Hour)+time.Minute-time.Millisecond, 720*time.Hour)),
 		},
 	}

